src/auth/controllers/auth.controller.ts

```
@@ -0,0 +1,105 @@
+import {
+  Body,
+  Controller,
+  Get,
+  HttpStatus,
+  Post,
+  Res,
+  UseGuards,
+} from '@nestjs/common';
+import { LoginUserDto } from '../dto/login-user.dto';
+import { AuthService } from '../services/auth.service';
+import {
+  ACCESS_TOKEN_COOKIE_NAME,
+  REFRESH_TOKEN_COOKIE_NAME,
+  REFRESH_TOKEN_EXPIRATION_S,
+} from '../constants/auth.constants';
+import { JwtRefreshAuthGuard } from '../guards/jwt-refresh.guard';
+import { CreateUserDto } from '../dto/create-user.dto';
+import { JwtAuthGuard } from '../guards/jwt.guard';
+import { CurrentUser } from '../decorators/current-user.decorator';
+import { UserInfo } from '../interfaces/user-info.interface';
+
+@Controller('auth')
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  private static setAuthCookies(res, tokens) {
+    res.cookie(ACCESS_TOKEN_COOKIE_NAME, tokens.access_token, {
+      httpOnly: true,
+    });
+    res.cookie(REFRESH_TOKEN_COOKIE_NAME, tokens.refresh_token, {
+      httpOnly: true,
+      maxAge: REFRESH_TOKEN_EXPIRATION_S * 1000,
+    });
+  }
+
+  @Post('login')
+  async login(@Body() request: LoginUserDto, @Res({ passthrough: true }) res) {
+    const email = request.email;
+    const password = request.password;
+    const user = await this.authService.validateUser(email, password);
+    if (user) {
+      const tokens = await this.authService.generateTokens(user);
+
+      AuthController.setAuthCookies(res, tokens);
+
+      return user;
+    }
+
+    // TODO: log and return a standardized error response
+    res.status(HttpStatus.UNAUTHORIZED).send({
```

Can we throw an error instead in the `validateUser` method?

---

Yeah, and let's consider using a homogenous way of handling errors and error responses across the whole application

---

Yes, let's do that from early on and use it across our service.